### PR TITLE
CMake-based v3.2.1

### PR DESCRIPTION
**Pull request by jowr**
Changes in this PR:
- Fixed the compilation issues with the ExternalMedia library
- Added a download link for CoolProp v5.1.1
- Requires CMake for compilation, but has been tested with the old CMake from the OMDev environment
- Some of the examples work with OpenModelica, but the topics discussed in #24 also appear here.

General comments:
- There is a new PR coming up that fixes the documentation, please only review the CompilationHowTo help file.
- Maybe we can merge this to the master branch since it is only a fix for version v3.2.1. It is not related to v3.3.0

### CODE REVIEW

**Comment by jowr** [comment]
In case anyone is interested: The binaries produced by https://github.com/jowr/ExternalMedia/actions/runs/465940532 can be found here:
https://ipudk-my.sharepoint.com/:u:/g/personal/jowr_ipu_dk/ET6Lc5G_B6JLtHDMfDOwa9kBHKzZ3aiHc-mdIKn1hZe1MA?e=5dj5nW

**Comment by jowr** [comment]
Just for the records, this PR potentially fixes #5, #7, #8, #14, #15, #21, #22. 

However, the problems with OpenModelica (#9, #10 and #24) seem to be caused by the translation step and are not addressed here.

**Comment by jowr** [comment]
Dear @casella - could you please share your thoughts on this contribution? It closes a lot of issues and I have the next PR ready to fix the documentation, but this one should be merged first... Let me know if you want anything modified.

File: Projects/BuildLib-CMake.bat [hunk 1]
```diff
@@ -7,26 +7,6 @@ REM    requires that you know the name of your CMake generator, if
...
-::set CMake_generator="Visual Studio 14 2015 Win64"
-
-set EXTERNALS="..\externals"
-if not exist "%EXTERNALS%" (mkdir "%EXTERNALS%")
-pushd "%EXTERNALS%"
-set CP_SRC=!CD!\CoolProp.git
-popd 
-
 set BUILD_DIR="build"
 if not exist "%BUILD_DIR%" (mkdir "%BUILD_DIR%")
```
Comment (beutlich) [inline Projects/BuildLib-CMake.bat:11]: I build dir exists and I previously built MSVC for "x64" a subesequent build for "Win32" will use the same build dir and fails.

**Review by beutlich — COMMENTED** [review]
Successfully tested that build works with VS2010.

File: Projects/CMakeLists.txt [hunk 2]
```diff
@@ -33,20 +33,43 @@ if (MSVC)
...
+  list (APPEND INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/../externals/CoolProp.git")
+  list (APPEND INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/../externals/CoolProp.git/include")
+  list (APPEND INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/../externals/CoolProp.git/externals/msgpack-c/include")
+  list (APPEND INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/../externals/CoolProp.git/externals/cppformat")
+else()
+  list(REMOVE_ITEM LIB_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/Sources/coolpropsolver.cpp")
+endif()
 
 ## We use CMake to handle the dependency since the primary VCS for 
 ## ExternalMedia still is SVN.
```
[the remark this answers is not in the document]
Comment (jowr) [reply Projects/CMakeLists.txt:75]: Yep, I am going to remove it.

File: Projects/CompilationHowTo.txt [hunk 3]
```diff
@@ -4,68 +4,36 @@ This directory contains the C/C++ source files in the Sources directory
 and script files to build the library using different Modelica tools,
 operating systems, and C/C++ compilers.
 
+Instructions below build the static gcc library and copy it and the
+externalmedia.h header files in the Resource directories of the
```
[the remark this answers is not in the document]
Comment (jowr) [reply Projects/CompilationHowTo.txt:8]: Please do not waste time on the docs. I have a new PR lined up that fixes the docs, but this one has to be merged first.

File: Projects/CompilationHowTo.txt [hunk 4]
```diff
@@ -4,68 +4,36 @@ This directory contains the C/C++ source files in the Sources directory
 and script files to build the library using different Modelica tools,
 operating systems, and C/C++ compilers.
 
+Instructions below build the static gcc library and copy it and the
```
[the remark this answers is not in the document]
Comment (jowr) [reply Projects/CompilationHowTo.txt:7]: Please do not waste time on the docs. I have a new PR lined up that fixes the docs, but this one has to be merged first.

File: Projects/CompilationHowTo.txt [hunk 5]
```diff
@@ -4,68 +4,36 @@ This directory contains the C/C++ source files in the Sources directory
 and script files to build the library using different Modelica tools,
 operating systems, and C/C++ compilers.
 
+Instructions below build the static gcc library and copy it and the
+externalmedia.h header files in the Resource directories of the
+Modelica packages, so it can be used right away by just loading the
+package in you Modelica software.
```
[the remark this answers is not in the document]
Comment (jowr) [reply Projects/CompilationHowTo.txt:10]: Please do not waste time on the docs. I have a new PR lined up that fixes the docs, but this one has to be merged first.

File: Projects/BuildLib-CMake.bat (hunk 1 above)
Comment (jowr) [reply Projects/BuildLib-CMake.bat:11]: Yes, we may want to remove the script since the whole installation process now only needs 2 CMake commands.

Comment (beutlich) [reply Projects/BuildLib-CMake.bat:11]: Indeed, I noticed when checking the CI part which is pretty impressive,

**Review by casella — APPROVED** [review]
@jowr, sorry for the delay and thank you for your patience. Now I have this think on the table.

I'll merge this and test it, and then report.

Thank you for the hard work!

**Comment by casella** [comment]
@jowr, now that I look closer, I guess this PR was meant to be merged on master, not on v.3.3.0-dev, right? After merging it, v.3.3.0-dev is mostly broken. We probably need to revert it.

Can we have a short webmeeting next week to coordinate? Now I finally have time to devote to ExternalMedia, and I can also get help from @fedetft, my local linux guru. I hope you also have some time now, so we can get this through

Thanks!

**Pull request merged**